docs/hardware_keyboard_guidelines.md

```diff
@@ -142,13 +144,14 @@ The `rules.mk` file can also be placed in a sub-folder, and its reading order is
     * `keyboards/top_folder/sub_1/sub_2/rules.mk`
       * `keyboards/top_folder/sub_1/sub_2/sub_3/rules.mk`
         * `keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/rules.mk`
-          * `keyboards/top_folder/keymaps/a_keymap/rules.mk`
           * `users/a_user_folder/rules.mk`
-        * `keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/post_rules.mk`
-      * `keyboards/top_folder/sub_1/sub_2/sub_3/post_rules.mk`
-    * `keyboards/top_folder/sub_1/sub_2/post_rules.mk`
-  * `keyboards/top_folder/sub_1/post_rules.mk`
-* `keyboards/top_folder/post_rules.mk`
+            * `keyboards/top_folder/keymaps/a_keymap/rules.mk`
```

Comment:
In the following description, the nesting levels of SUB_X/rules.mk and SUB_X/post_rules.mk do not match.

* `keyboards/top_folder/rules.mk`
  * `keyboards/top_folder/sub_1/rules.mk`
    * `keyboards/top_folder/sub_1/sub_2/rules.mk`
      * `keyboards/top_folder/sub_1/sub_2/sub_3/rules.mk`
        * `keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/rules.mk`
          * `users/a_user_folder/rules.mk`
            * `keyboards/top_folder/keymaps/a_keymap/rules.mk`
          * `keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/post_rules.mk`
        * `keyboards/top_folder/sub_1/sub_2/sub_3/post_rules.mk`
      * `keyboards/top_folder/sub_1/sub_2/post_rules.mk`
    * `keyboards/top_folder/sub_1/post_rules.mk`
  * `keyboards/top_folder/post_rules.mk`
* `users/a_user_folder/post_rules.mk`
* `common_features.mk`

I feel it would be better to match the nest levels as follows.

* `keyboards/top_folder/rules.mk`
  * `keyboards/top_folder/sub_1/rules.mk`
    * `keyboards/top_folder/sub_1/sub_2/rules.mk`
      * `keyboards/top_folder/sub_1/sub_2/sub_3/rules.mk`
        * `keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/rules.mk`
          * `users/a_user_folder/rules.mk`
          * `keyboards/top_folder/keymaps/a_keymap/rules.mk`
        * `keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/post_rules.mk`
      * `keyboards/top_folder/sub_1/sub_2/sub_3/post_rules.mk`
    * `keyboards/top_folder/sub_1/sub_2/post_rules.mk`
  * `keyboards/top_folder/sub_1/post_rules.mk`
* `keyboards/top_folder/post_rules.mk`
* `users/a_user_folder/post_rules.mk`
* `common_features.mk`